### PR TITLE
Prepare 2.0.0 release

### PR DESCRIPTION
**Pull request by felixfontein**
See #15.

Tests will likely fail until antsibull-docs's and antsibull's `main` branches support antsibull-core 2.x.y.

### CODE REVIEW

**Comment by felixfontein** [comment]
PRs to make antsibull and antsibull-docs accept antsibull-core 2.0.0 are https://github.com/ansible-community/antsibull-docs/pull/78 and https://github.com/ansible-community/antsibull/pull/463.

File: src/antsibull_core/dependency_files.py [hunk 1]
```diff
@@ -38,11 +38,8 @@ def parse_pieces_file(pieces_file: str) -> List[str]:
 
     decoded_contents = contents.decode('utf-8')
     # One collection per line, ignoring comments and empty lines
-    # TODO: PY3.8:
-    # collections = [c for line in decoded_contents.split('\n')
-    #                if (c := line.strip()) and not c.startswith('#')]
-    collections = [line.strip() for line in decoded_contents.split('\n')
-                   if line.strip() and not line.strip().startswith('#')]
+    collections = [c for line in decoded_contents.split('\n')
+                   if (c := line.strip()) and not c.startswith('#')]
```
[the remark this answers is not in the document]
Comment (felixfontein) [reply src/antsibull_core/dependency_files.py:42]: I think it's best to keep the current behavior.

File: src/antsibull_core/compat.py [hunk 2]
```diff
@@ -5,57 +5,34 @@
...
-    # As long as https://bugs.python.org/issue46080 is not fixed, still use vendored version so
-    # that users can actually use `--help` for affected subcommands. Once that is fixed and
-    # included in released Python versions, we can adjust the sys.version_info check above.
-    #
-    #     from argparse import BooleanOptionalAction
-    from .vendored._argparse_booleanoptionalaction import BooleanOptionalAction
+
+best_get_loop = asyncio.get_running_loop
+asyncio_run = asyncio.run
+create_task = asyncio.create_task
```
[the remark this answers is not in the document]
Comment (felixfontein) [reply src/antsibull_core/compat.py:41]: 47f67c8cc717e9c1e16518be3e51c451ce915a83.

File: src/antsibull_core/compat.py [hunk 3]
```diff
@@ -5,57 +5,34 @@
 # SPDX-FileCopyrightText: 2020, Ansible Project
 """Compat for older versions of Python."""
 
+import argparse
 import asyncio
 import sys
+import typing as t
 
+from importlib import metadata
```
[the remark this answers is not in the document]
Comment (felixfontein) [reply src/antsibull_core/compat.py:17]: I removed the usage in 47f67c8cc717e9c1e16518be3e51c451ce915a83, but removing it from this file is a breaking change that should have deprecation first.

File: src/antsibull_core/compat.py [hunk 4]
```diff
@@ -5,57 +5,34 @@
 # SPDX-FileCopyrightText: 2020, Ansible Project
 """Compat for older versions of Python."""
```
[the remark this answers is not in the document]
Comment (felixfontein) [reply src/antsibull_core/compat.py]: 6977193154ec70ea0be451861f0b2315bd905391.

[the remark this answers is not in the document]
Comment (felixfontein) [reply src/antsibull_core/compat.py]: Meh, I didn't think too far apparently... I think we should keep this module, just deprecate the things in it we no longer need. I've updated this in 459479ce675f628dade333fb9ed071144623666e.

We will need `BooleanOptionalAction` some time longer (until we deprecate Python 3.9 and 3.10), and potentially we need to add more stuff in the future.

File: src/antsibull_core/compat.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gotmax23) [reply src/antsibull_core/compat.py:43]: That commit adds duplicate asyncio imports.

File: pyproject.toml [hunk 5]
```diff
@@ -29,11 +29,9 @@ classifiers = [
 "Bug tracker" = "https://github.com/ansible-community/antsibull-core/issues"
 
 [tool.poetry.dependencies]
-python = "^3.6.1"
-aiocontextvars = {version = "*", python = "~3.6"}
+python = "^3.9.0"
```
Comment (gotmax23) [inline pyproject.toml:32]: Do we need a changelog fragment for bumping the minimum Python version?

Comment (felixfontein) [reply pyproject.toml:32]: Oh, yes, definitely...

Comment (felixfontein) [reply pyproject.toml:32]: d093c71b252c8fd7bf217b05dcd253e8b9069fc3.

**Comment by gotmax23** [comment]
We should also update https://github.com/ansible-community/antsibull-core#versioning-and-compatibility but that can be separate.

**Comment by felixfontein** [comment]
@gotmax23 thanks for reviewing this!

Let's do more polishing in follow-up PRs, so that CI in antsibull and antsbull-docs can already run against this merged PR.

**Comment by gotmax23** [comment]
Hah, I was just pushing the merge button and then you beat me to it :).

**Comment by gotmax23** [comment]
> @gotmax23 thanks for reviewing this!

Sure :)

**Pull request merged**